### ISSUE TITLE
Help: Aisler Errors in Smart Testing

### ISSUE THREAD

**Issue by kamronbatman** [kinx-project/kint#65]
Hey everyone,

I decided to venture into DIY hardware and this is my first project so I need some guidance.

I ordered boards based on the newest specs and it looks like they came back with errors during testing. Does this mean I should ask them to manufacture them again since they will be defective? Or is this ok?

Thanks!

![20230427_130836.jpg](https://user-images.githubusercontent.com/3953314/234979340-003f3f68-a75d-4576-907c-449c96e28d31.jpg)

**Comment by stapelberg** [comment]
> I ordered boards based on the newest specs and it looks like they came back with errors during testing. Does this mean I should ask them to manufacture them again since they will be defective? Or is this ok?

The component in question is the solder jumper, so I’m leaning towards saying that this is okay and nothing to worry about. 

I’m not entirely sure why you’re getting this error and not others (or maybe others are getting the error, too?). Maybe this changed with KiCad 6, or maybe we should be doing something differently in our design files… Maybe someone with more KiCad experience can weigh in.

**Comment by kamronbatman** [comment]
Here is the response from support:

![Screenshot_20230428_114332_Gmail.jpg](https://user-images.githubusercontent.com/3953314/235228308-750a1d31-aa4e-4bc3-95ba-9713900e9e8c.jpg)

**Comment by stapelberg** [comment]
I wonder if this issue was introduced with commit https://github.com/kinx-project/kint/commit/afac7593b2d746b0df2951f4b0d2ef72b4ce3b4c (cc @cincodenada)

If you want a known-working revision, v2021-04-25 is the latest revision that I have in my keyboards. Sorry for the trouble with this.

**Comment by stapelberg** [comment]
To be quite honest, even after re-reading the support reply, I’m not sure if there really is a problem with the boards. 

In the reply, JP4, JP5 and JP6 are listed, and those are all the connected-by-default solder jumpers on the board. I’m still suspecting that either their tool gets tripped up by solder jumpers, or that we have somehow incorrectly designed/placed the solder jumpers.

A connection between J5/1 and U1/7 (the “fault” the e-tester found) is expected, so maybe no correction is necessary at all and the board just works.

But, yes, in case you do need to correct something, I think you could just cut traces close to the Teensy pins on the PCB. I would give it a try (no need to order a new revision just yet), but be sure you have more than one Teensy in case something goes wrong :)

**Comment by kamronbatman** [comment]
I ordered 3 of everything, but I am thinking about ordering removable pin headers for the teensy, so if I have to cut/desolder, it's only pin headers. 🤔

**Comment by aleb** [comment]
The boards are fine. I also noticed this error before. Aisler's software expects these "nets" to be disconnected. The idea is that if they are connected then they should be a single "net".

**Comment by cincodenada** [comment]
Oh sorry, I had drafted a reply but never finished it! The others are correct, yes, the boards should function just fine - the net-ties here are indeed intentional, the jumpers are intended to be connected these two nets in the normal case, if you were working with a Teensy 2 you would cut them to disconnect.

I'm not sure if there is a way to notate that in a way that Aisler will be happy with, I thought that the new version of KiCad had improved net-tie support, but I'm not familiar enough with the details of those exports to know - are you uploading the KiCad files directly to Aisler, or generated Gerbers?

**Comment by cincodenada** [comment]
If there's an easy way to notate this in a way that will make Aisler's checker happy, that would be optimal, otherwise we might want to add a note to the README about ignoring errors about shorts in JP4/5/6

**Comment by kamronbatman** [comment]
yeah, I am not sure how to either, but I can ask their support. I uploaded the [kint.kicad_pcb](https://github.com/kinx-project/kint/blob/main/kicad/kint.kicad_pcb) file directly. Maybe I was supposed to send them other files too, but I didn't see a way to do that, so I am not sure.

**Comment by stapelberg** [comment]
> Maybe I was supposed to send them other files too, but I didn't see a way to do that, so I am not sure.

No, the advantage of Aisler and other modern manufacturers is that you only need to upload one kicad file and they do the remaining processing.

**Comment by stapelberg** [comment]
Let’s leave this issue open for easier discoverability. And, ideally someone could contribute a fix to make the design just work with Aisler’s checker :)

**Comment by impaktor** [comment]
(github allows pinning an issue. Not sure if it only works for open or also possible for closed issues).
